Excel: A sound is played when typing into protected cells rather than speaking typed characters

michaelDCurran opened the pull request:
Fixes #6570

michaelDCurran committed:
Excel: play the default sound rather than speaking typed characters when typing into protected cells.

michaelDCurran commented:
@feerrenrut: review please.

feerrenrut commented on source/NVDAObjects/window/excel.py:
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
 
+	def event_typedCharacter(self,ch):
+		# # 6570: You cannot type into protected cells.
+		# A part from speaking characters being miss-leading, Office 2016 protected view doubles characters as well.
```
minor: "Apart" rather than "A part"

feerrenrut commented on source/NVDAObjects/window/excel.py:
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
 		return states
 
+	def event_typedCharacter(self,ch):
+		# # 6570: You cannot type into protected cells.
```
minor: there is a space between the hash symbol and the issue number.

feerrenrut commented on source/NVDAObjects/window/excel.py:
```diff
@@ -1155,6 +1156,15 @@ def _get_states(self):
+		# # 6570: You cannot type into protected cells.
+		# A part from speaking characters being miss-leading, Office 2016 protected view doubles characters as well.
+		# Therefore for any non-control char on protected cells, play the default sound rather than speaking the character
+		if ord(ch)>26 and controlTypes.STATE_UNLOCKED not in self.states and controlTypes.STATE_PROTECTED in self.parent.states: 
```
It would be helpful if 26 was a named value. Is it the first printable character code?

michaelDCurran committed:
Excel: address review comments

michaelDCurran commented:
How about this @feerrenrut? I addressed the comment issues, but I changed the numeric value into a space. I can add a constant if you think it is really necessary...

feerrenrut reviewed (approved):
Looks good to me.
